lib/prefill.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+module Declarations
+  module Prefills
+    class MonthlyPrefillService < ApplicationService
+      SENTRY_OWNER = Constants::SentryOwners::DECLARATIONS
+
+      Error = Exceptions::AppInternalError.with_scope('declarations.prefills.prefill_service')
+
+      class UnsupportedDeclaration < Error; end
+
+      private
+
+      def initialize(declaration:)
+        @declaration = declaration
+        @company = @declaration.company
+      end
+
+      def call
```

Comment:
The `call` method should be structured as a list of actions and avoid complex conditions. The current implementation with a return statement inside a condition can be refactored for better readability.
```suggestion
 def call
   if find_prefill
     prefill_declaration
   end
   @declaration
 end
```